camel/societies/babyagi_playing.py

```diff
@@ -0,0 +1,287 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from collections import deque
+from typing import Dict, List, Optional
+
+from colorama import Fore
+
+from camel.agents import (
+    ChatAgent,
+    TaskCreationAgent,
+    TaskPrioritizeAgent,
+    TaskSpecifyAgent,
+)
+from camel.agents.chat_agent import ChatAgentResponse
+from camel.generators import SystemMessageGenerator
+from camel.messages import BaseMessage
+from camel.prompts import TextPrompt
+from camel.typing import RoleType, TaskType
+from camel.utils import print_text_animated
+
+
+class BabyAGI:
+    r"""BabyAGI Agent.
+
+    Args:
+        assistant_role_name (str): The name of the role played by the
```

Comment:
Is it copy-pasted from somewhere?

Reply:
Yeah, it's copied from role_playing.py. But we still need this parameter in Babyagi.

Reply:
@lightaime this parameter list is getting out of hand. It should not be that long. And also now it gets copy-pasted around. This will lead to problems in the future. I will not impede this change to keep velocity this time around, but we need to rethink the architecture.